File: src/decoders/partner-ard.ts
import type { PartnerARD } from "~/models";
import { decodePartnerARDWallet } from "./partner-ard-wallet";
import { decodePartner } from "./partner";

export const decodePartnerARD = (partner: any): PartnerARD => {
  return {
    ...decodePartner(partner),
    canRefreshData: partner.avecActualisation,
    wallets: partner.porteMonnaie.V.map(decodePartnerARDWallet)
  };
};

File: library/src/commonMain/kotlin/decoders/partner-ard.kt
package decoders

import kotlinx.serialization.json.*
import models.PartnerARD

fun decodePartnerARD (partner: JsonObject): PartnerARD {
    return PartnerARD(
        sso = decodePartner(partner).sso,
        canRefreshData = partner["avecActualisation"]!!.jsonPrimitive.boolean,
        wallets = partner["porteMonnaie"]!!.jsonObject["V"]!!.jsonArray.map { decodePartnerARDWallet(it.jsonObject) }
    )
}